tests/governance.py

```diff
@@ -229,6 +229,13 @@ def read_service_principal():
     return network
 
 
+@reqs.description("Test ack state digest updates")
+def test_ack_state_digest_update(network, args):
+    for _ in range(50):
```

Comment:
Seems like an arbitrary number - is it necessary to try this that often?

Reply:
It is arbitrary, but users have reported in #2191 that they sometimes see validation failures even when they have an empty body/content-length zero. Since there is as far as I can tell no way for that to happen, this is all I can think to do to increase our chances of catching this issue if it exists. Ideas welcome.

Reply:
I see, thanks for the explanation! The only thing I can think of is to add random (small) delays on various operations so we get unusual execution sequences. Not sure that's really a factor here though.

Reply:
Come to think of it what we should test is mismatched content length and sizes probably.